core-bundle/src/EventListener/DataContainer/PageTypeOptionsListener.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\Event\FilterPageTypeEvent;
+use Contao\CoreBundle\Routing\Page\PageRegistry;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\DataContainer;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Security\Core\Security;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+/**
+ * @Callback(table="tl_page", target="fields.type.options")
+ */
+class PageTypeOptionsListener implements ServiceAnnotationInterface
+{
+    /**
+     * @var PageRegistry
+     */
+    private $pageRegistry;
+
+    /**
+     * @var Security
+     */
+    private $security;
+
+    /**
+     * @var EventDispatcherInterface|null
+     */
+    private $eventDispatcher;
+
+    public function __construct(PageRegistry $pageRegistry, Security $security, EventDispatcherInterface $eventDispatcher = null)
+    {
+        $this->pageRegistry = $pageRegistry;
+        $this->security = $security;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function __invoke(DataContainer $dc)
+    {
+        $options = array_unique(array_merge(array_keys($GLOBALS['TL_PTY']), $this->pageRegistry->keys()));
+
+        if (null !== $this->eventDispatcher) {
+            $options = $this->eventDispatcher
+                ->dispatch(new FilterPageTypeEvent($options, $dc))
+                ->getOptions()
+            ;
+        }
+
+        // Allow the currently selected option and anything the user has access to
+        foreach ($options as $k => $pageType) {
+            if ($pageType !== $dc->value && !$this->security->isGranted('contao_user.alpty', $pageType)) {
```

Comment:
This should be `ContaoCorePermissions::USER_CAN_ACCESS_PAGE_TYPE`, shouldn't it?

Reply:
Fixed in 0dea19a8229a1f0f3a0c5747349cce228cffda9c